code/parachain/frame/currency-factory/src/ranges.rs

```diff
@@ -111,64 +152,82 @@ impl<AssetId> Range<AssetId>
 where
 	AssetId: From<u128> + Saturating + Ord + Clone,
 {
+	/// Retruns the end `AssetId` of this `Range`.
 	pub fn end(&self) -> AssetId {
 		self.end.clone()
 	}
 
+	/// Returns the current `AssetId` of this `Range`.
 	pub fn current(&self) -> AssetId {
 		self.current.clone()
 	}
 
+	// Preset ranges
+	// Ranges should be made in chunks of the size `u32::MAX`.
+	// Meaning, range `n` will be from `u32::MAX * n + 1` to `u32::MAX * (n + 1)`.
+	// The range `0` to `u32::MAX` is used for assets defined by the runtime.
+
+	// REVIEW(connor): Range definitions could be simplified as a function of `n`. Should we
+	// simplify the code, or avoid abstracting this away.
+	// NOTE(connor): I plan on moving preconfigured ranges to the runtime config. More details to
+	// come in RFC.
+
+	/// Range for LP Tokens.
 	fn lp_tokens() -> Self {
 		Range {
-			current: AssetId::from(100_000_000_001_u128),
-			end: AssetId::from(200_000_000_000_u128),
+			current: AssetId::from((u32::MAX as u128).saturating_add(1)),
+			end: AssetId::from((u32::MAX as u128).saturating_mul(2)),
```

Comment:
Maybe use `checked_*().expect()` here and then have tests to assert they don't error

Reply:
(for all of these associated fns)

Reply:
Gonna look gross, but agreed

Reply:
could probably write a simple macro_rules! to make it a bit cleaner

Reply:
68ca44cdd41076ed144ec6df07e5ed65c6481899